### PR TITLE
feat: Add X-NC-Disable-Scheduling property to allow skipping scheduling

### LINKED ISSUE #48528: Add X-NC-Disable-Scheduling property to allow skipping scheduling deliveries
When syncing with a different Calendar sever that PUTs all events, including the already delivered scheduling events for ATTENDEEs, the scheduling logic shouldn't trigger local delivery again.

Add the X-NC-Disable-Scheduling property that will skip both the SchedulingPlugin and the IMipPlugin so users don't get notified twice.

### PR DESCRIPTION
**Pull request by SebastianKrupinski**
* Resolves: #48528

## Summary
Checks event for X-NC-Disable-Scheduling = 1 property and value and omits sending iMip messages

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: apps/dav/lib/CalDAV/Schedule/IMipPlugin.php [hunk 1]
```diff
@@ -95,8 +95,15 @@ public function beforeWriteContent($uri, INode $node, $data, $modified): void {
 	 * @return void
 	 */
 	public function schedule(Message $iTipMessage) {
-		// Not sending any emails if the system considers the update
-		// insignificant.
+
+		// do not send imip messages if external system already did
+		if ($iTipMessage->message?->VEVENT?->{'X-NC-DISABLE-SCHEDULING'}?->getValue() === '1') {
```
Comment (miaulalala) [inline apps/dav/lib/CalDAV/Schedule/IMipPlugin.php:100]: Can you change that to use true? I don't like a stringified integer for checks like these.

**Review by st3iny — COMMENTED** [review]
Looks good otherwise.

File: apps/dav/tests/unit/CalDAV/Schedule/IMipPluginTest.php [hunk 2]
```diff
@@ -890,4 +890,23 @@ public function testNoButtons(): void {
...
+		$event->add('X-NC-DISABLE-SCHEDULING', '1');
+		$message = new Message();
+		$message->method = 'REQUEST';
+		$message->message = $calendar;
+		$message->sender = 'mailto:gandalf@wiz.ard';
+		$message->senderName = 'Mr. Wizard';
+		$message->recipient = 'mailto:' . 'frodo@hobb.it';
+		
+		$this->plugin->schedule($message);
+		$this->assertEquals('1.0', $message->getScheduleStatus());
```
Comment (st3iny) [inline apps/dav/tests/unit/CalDAV/Schedule/IMipPluginTest.php:910]:
```suggestion
		$this->assertEquals('1.0;We got the message, but iMip message are disabled for this event', $message->scheduleStatus);
```

Just to make the test a bit more robust.

**Review by st3iny — APPROVED** [review]

**Review by miaulalala — APPROVED** [review]

**Pull request merged**